app/views/hosts/_form.html.erb

```diff
@@ -138,6 +138,11 @@
 
     <%= hidden_field_tag 'bare_metal_capabilities', @host.bare_metal_capabilities, :disabled => true %>
     <%= f.hidden_field :overwrite? %>
-    <%= submit_or_cancel f, @host.overwrite?, :cancel_path => (@host.new_record? ? hosts_path : host_path(:id => @host.name_was)) %>
+    <% resource_prev_url = resource_prev_url_with_search_filters
+      unless resource_prev_url
+        resource_prev_url = @host.new_record? ? hosts_path : host_path(:id => @host.name_was)
+      end
+    %>
+    <%= submit_or_cancel f, @host.overwrite?, :cancel_path => resource_prev_url %>
```

Comment:
I do not like assigning variables in the `erb`, I would prefer:
```suggestion
    <%= submit_or_cancel f, @host.overwrite?, :cancel_path => resource_prev_url_with_search_filters || (@host.new_record? ? hosts_path : host_path(:id => @host.name_was)) %>
```
If you would disagree, to have the line shorter, I would at least use:
```ruby
resource_prev_url ||= @host.new_record? ? hosts_path : host_path(:id => @host.name_was)
```
instead of the unless statement.